### PR TITLE
feat: support addon features for paygate mini

### PR DESCRIPTION
**Pull request by raquelmsmith**
## Problem

PayGate Mini only worked if the feature in question was for a primary product - not for an addon feature.

## Changes

Updates it so that can detect features that belong to addons, and give a proper message in the component

image

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Does this work well for both Cloud and self-hosted?

Yes

## How did you test this code?

🐭

### CODE REVIEW

File: frontend/src/scenes/plugins/tabs/batch-exports/BatchExportsTab.tsx [hunk 1]
```diff
@@ -6,9 +6,7 @@ import { AvailableFeature } from '~/types'
 export function BatchExportsTab(): JSX.Element {
     return (
         <>
-            <PayGateMini feature={AvailableFeature.DATA_PIPELINES}>
-                <></>
-            </PayGateMini>
```
Comment (tiina303) [inline frontend/src/scenes/plugins/tabs/batch-exports/BatchExportsTab.tsx:11]: Thank you ❤️  this is much better :)

File: frontend/src/lib/components/PayGateMini/PayGateMiniButton.tsx [hunk 2]
```diff
@@ -5,22 +5,28 @@ import { billingProductLogic } from 'scenes/billing/billingProductLogic'
...
     onClick?: () => void
 }): JSX.Element => {
     const { isPlanComparisonModalOpen } = useValues(billingProductLogic({ product }))
     const { toggleIsPlanComparisonModalOpen } = useActions(billingProductLogic({ product }))
 
+    // We know that the product is a BillingProductV2Type because that's the only
+    // type that can be used here from the PayGateMini component. But TypeScript doesn't
+    // know that, so we need to cast it to the correct type, and it's simpler to do this
+    // here than in the PayGateMini component.
+    const typedProduct = product as BillingProductV2Type
```
Comment (tiina303) [inline frontend/src/lib/components/PayGateMini/PayGateMiniButton.tsx:28]: I'm confused why did you change the product type at the top then?

**Review by tiina303 — COMMENTED** [review]
🙇‍♀️

File: frontend/src/lib/components/PayGateMini/PayGateMiniButton.tsx (hunk 2 above)
Comment (raquelmsmith) [reply frontend/src/lib/components/PayGateMini/PayGateMiniButton.tsx:28]: Because if I didn't then I'd get a type error in the PayGateMini component. But in that component I _can_ have either an addon or a regular product type. TBH idk why the addon type is different from the regular product type...
